tesla_powerwall/api.py

```diff
@@ -21,10 +19,7 @@ def __init__(
         verify_ssl: bool = False,
         disable_insecure_warning: bool = True,
     ) -> None:
-        if disable_insecure_warning:
-            disable_warnings(InsecureRequestWarning)
-
-        self._endpoint = self._parse_endpoint(endpoint)
+        self._endpoint = URL(self._parse_endpoint(endpoint))
```

Comment:
```suggestion
        self._endpoint = URL(endpoint).with_scheme("https").with_path("api")
```

yarl makes the URL handling much simpler and makes the `_parse_endpoint` method obsolete.